src/execution/operator/persistent/physical_export.cpp

```diff
@@ -118,8 +114,10 @@ void PhysicalExport::GetChunkInternal(ExecutionContext &context, DataChunk &chun
 	// write the load.sql file
 	// for every table, we write COPY INTO statement with the specified options
 	stringstream load_ss;
+	idx_t i = 0;
 	for (auto &table : tables) {
-		WriteCopyStatement(fs, load_ss, (TableCatalogEntry *)table, *info, function);
+		WriteCopyStatement(fs, load_ss, (TableCatalogEntry *)table, *info, exported_tables.at(i), function);
```

Comment:
The exported_tables seems to rely on the tables scan to output the tables in the same order every time. While that is true currently, that assumption might get violated in the future. What might be better is something along the lines of e.g.

```cpp
struct BoundExportData : public ParseInfo {
    unordered_map<TableCatalogEntry *, ExportedTableData> data;
};
```

Then you have a direct map of table entry -> export data, and you can check that all of the tables are present.

Reply:
Just to be sure I understand, if we gather table info on the bind_export.cpp then we don't need to re-gather the same table info on the physical_export.cpp. Is that correct?

physical_export.cpp:90
```C++
schema->Scan(context.client, CatalogType::TABLE_ENTRY, [&](CatalogEntry *entry) {
	if (entry->type == CatalogType::TABLE_ENTRY) {
		tables.push_back(entry);
	} else {
		views.push_back(entry);
	}
})
```

Reply:
Yes, indeed. The export lives in a transaction, and a client can only run one query at a time, so necessarily the set of tables visible to a client cannot change in the middle of a query.